tests/phpunit/tests/test-admin-filters-term.php

```diff
@@ -414,6 +414,47 @@ function test_new_default_category() {
 		$this->assertEqualSets( array( 'en', 'fr', 'de', 'es' ), array_keys( $translations ) );
 	}
 
+	function test_new_default_category_with_empty_translations() {
+		$old = self::$polylang->model->term;
+		self::create_language( 'pt_PT' );
+		$default = (int) get_option( 'default_category' );
+
+		self::$polylang->model->term = $this->getMockBuilder( 'PLL_Translated_Term' )
+			->setMethods(
+				array(
+					'get_translations',
+				)
+			)
+			->setConstructorArgs(
+				array(
+					&self::$polylang->model,
+				)
+			)
+			->getMock();
+		self::$polylang->model->term->method( 'get_translations' )
+			->will(
+				$this->returnCallback(
+					function( $id ) {
+						$test = new PLL_Translated_Term( self::$polylang->model );
+						$default = (int) get_option( 'default_category' );
+						if ( $id === $default ) {
+							return null;
+						}
+						return ( $test->get_translations( $id ) );
+					}
+				)
+			);
+
+
+		self::$polylang->model->create_default_category( self::$polylang->model->get_language( 'pt' )->slug );
+
+		self::$polylang->model->term = $old;
+
+		$translations = self::$polylang->model->term->get_translations( $default );
+
+		$this->assertEqualSets( array( 'en', 'pt' ), array_keys( $translations ) );
+	}
+
```

Comment:
It looks to me that https://github.com/polylang/polylang/blob/2.8.4/include/model.php#L398-L405 is dead code. Indeed `$translations = $this->term->get_translations( $default );` should at least return `$default`, due to https://github.com/polylang/polylang/blob/2.8.4/include/translated-object.php#L196-L198.

So you succeeded to execute these lines by mocking `PLL_Translated_Term::get_translations()` but this should not be representative of the behavior in production. I rather propose not to add this test and later remove the corresponding code in your refactoring.